### DISCUSSION TITLE
Who’s using Phlex?

### DISCUSSION THREAD

**Discussion by joeldrapper** [yippee-fun/phlex#530] [General]
Hey everyone 👋 

I’m _really_ interested in hearing about who’s using Phlex — especially `phlex-rails` in production. Let me know in the comments.

**Comment by joelmoss** [comment]
I'm currently building a new Rails app completely using Phlex, but it won't be in production until later this year.

**Comment by ianterrell** [comment]
I am super tempted to begin a UI rewrite with it in our main app suite, but am anxious about support and support around form helpers and Turbo, which ViewComponent also struggles with.

**Reply by joeldrapper** [reply]
That's understandable. We do have full compatibility with the Rails form helpers now, but it's still early and there could be edge cases we haven't explored. One thing to keep in mind is you can always render a regular partial from a Phlex component if you need to keep one part like a form in ERB for some reason.

I think the turbo compatibility for both Phlex and ViewComponent is resolved with this PR to `turbo-rails`. https://github.com/hotwired/turbo-rails/pull/433 Is there anything in particular that isn’t working as expected in turbo?

**Reply by ianterrell** [reply]
Thanks for pointing me to that PR — I will explore where we had issues and undo our workaround and see if the behavior works as I expect (or continue to update my expectations).

One experiment I was doing with Phlex and forms may be ill advised, but I was using a custom FormBuilder to render Phlex components that then leveraged the original Rails helpers. It is a bit convoluted but led to a fairly nice call site. I'll see if I can build a small example for another discussion.

**Reply by joeldrapper** [reply]
`phlex-rails` will eventually provide a base form component that you can use like a custom form builder, only it will be a component, with all the performance and safety benefits. I believe the `form_with` and `form_for` adapters for Phlex do support custom form builders, though.

**Comment by hms** [comment]
We're using phlex strategically -- converting more complex partials, especially with turbo_stream to Phlex.

We have a small app that supports doctors helping patients with reaching Shared Decision Making and Medical Informed Consent.  We're currently supporting about 60 users and growing slowly.  We'll be releasing the version that's using Phlex with our next major upgrade in the next couple of months.

**Reply by joeldrapper** [reply]
That’s awesome! I’d love to hear how you get on over the next few months.

**Comment by wdiechmann** [comment]
We are currently in the process of rewriting a Time & Attendance app using Phlex (coming from view_components) - and as witnessed in 1-2 discussions here 😓 it's not been without challenges -- but thanks to you @joeldrapper I am very happy to share that not only will we go forward (are going already) with this, but Phlex will be our _weapons of choice_ 

And if I may on a personal note comment that I cannot wait to see when you will present us with the spoils of your awaited sign-up bonus from GitHub (or however) - 'cause in my point of view Phlex is way more fun, and easy, and faster than ViewComponents, et al!!!

Lots of 👏

**Reply by joeldrapper** [reply]
Thanks so much, that means a lot. ❤️

**Comment by hangsu** [comment]
We're using Phlex for about 60% of our views in production here at clearscope.io, serving 1,000s of paying customers.

It's been a huge success so far thanks to @joeldrapper's commitment to the project, and we're looking forward to being 100% on Phlex within months.

I'm happy to share our experience with anyone so feel free to reach out.

**Comment by hmaddocks** [comment]
I'm using Phlex in two rails apps. The first (https://www.switchhub.nz) I'm migrating from Rails erb templates to Phlex and the second, not in production yet, has been built with Phlex from the beginning. It's going great for most things. Turbo::Broadcastable tripped me up recently but I have just figured out a solution https://henry.bearblog.dev/rendering-phlex-components-with-turbo-streams/
Thanks for your work on this.

**Comment by seanhogge** [comment]
I'm using phlex-rails to rebuild a Rails 6 app that serves as an intranet to an OEM. It does dashboarding, CRUD, and everything you'd expect. It leverages Turbo Morphing and Turbo Streams to show "live" pages, chat-like comment sections, and componentizes all the things.

It's been a shallower learning curve than I expected. All struggles have revolved around helpers such as Pagy and authenticity tokens (and maybe exposes my own ignorance of those systems more than anything). The standard stuff like managing Tailwind classes has been delightful.

For me, it's been surprising how useful it is to solve view problems with Ruby. A view being a class that can use methods and modules has been liberating. No longer needed to context switch, even to something as simple as HTML is pleasant even if it's not actually any easier. Plus my linter can actually format Ruby code. HTML; not so much.

**Comment by UnderpantsGnome** [comment]
We are using Phlex to replace our current React UI with Hotwire etc. on [VivoPoint](https://www.vivopoint.com/). We've run into some quirks using the [bootstrap_form](https://github.com/bootstrap-ruby/bootstrap_form) gem, but nothing major.

So far so good, so much better than ERB soup. Keep up the good work. 🙂

**Comment by janpeterka** [comment]
I recently started new MVP, and when choosing stack, I decided to use RubyUI for fast components, which also means using Phlex.
I'm not yet fully there, using default generated layout (it would be great if default layout - same as with `rails new` -  could be generated by Phlex) and simple_form in partials. 

I was using `slim` before, so I was used to pretty minimal view code, but what I really like is how writing views in Ruby leads me more to component/small methods way, same as any other Ruby code. In html I was used to having pretty long code because "thats how it is done", but in Ruby I have strong bias for small, modular, simply refactorable parts.

I just added Literal today, and I will be using that bit more - in many cases, my initialization is one or two variables, and declaring `prop` is just much simpler.

So thanks for this interesting and fun new way of creating app!
